File: check_omdb.js
const fetch = require('node-fetch');

async function test() {
    const res = await fetch('http://116.203.230.103/api/details/movie/tt1375666');
    const data = await res.json();
    console.log("Type of omdb:", typeof data.omdb);
    console.log("Is array?", Array.isArray(data.omdb));
    if (typeof data.omdb === 'string') {
        console.log("It's a string! Let's try parsing it.");
        try {
            const parsed = JSON.parse(data.omdb);
            console.log("Parsed ratings:", parsed.Ratings);
        } catch(e) {
            console.log("Parse error");
        }
    } else {
        console.log("Ratings:", data.omdb ? data.omdb.Ratings : 'omdb is null');
    }
}
test();

File: check_omdb.py
import urllib.request
import json

req = urllib.request.urlopen("http://116.203.230.103/api/details/movie/tt1375666")
data = json.loads(req.read())
omdb = data.get('omdb')

print(f"Type of omdb: {type(omdb)}")
if isinstance(omdb, str):
    print("It's a string! Let's try parsing it.")
    try:
        parsed = json.loads(omdb)
        print("Parsed ratings:", parsed.get('Ratings'))
    except:
        print("Parse error")
else:
    print("Ratings:", omdb.get('Ratings') if omdb else "omdb is null")
